addons/mrp/models/mrp_bom.py

```diff
@@ -737,10 +717,8 @@ class MrpBomByproduct(models.Model):
     product_qty = fields.Float(
         'Quantity',
         default=1.0, digits='Product Unit of Measure', required=True)
-    product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
     product_uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True,
-                                     compute="_compute_product_uom_id", store=True, readonly=False, precompute=True,
-                                     domain="[('category_id', '=', product_uom_category_id)]")
+                                     compute="_compute_product_uom_id", store=True, readonly=False, precompute=True)
```

Comment:
domains need to be checked between BoM and MO to avoid mismatch (i.e. last time I tested, you could select UoMs in the BoM that aren't selectable in the MO)